prow/gerrit/adapter/adapter.go

```diff
@@ -585,6 +585,32 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 			logger.WithField("lastUpdate", lastUpdate).Warnf("lastUpdate not found, falling back to now")
 		}
 
+		if change.Revisions != nil {
+			var hasTestOrRetestCommand bool
+			for _, message := range currentMessages(change, lastUpdate) {
+				if pjutil.RetestRe.Match([]byte(message.Message)) || pjutil.TestRe.Match([]byte(message.Message)) {
+					hasTestOrRetestCommand = true
+					break
+				}
+			}
+
+			var hasCodeChange bool
+			for _, rev := range change.Revisions {
+				if rev.Created.Time.After(lastUpdate) && rev.Kind != gerrit.NoCodeChange {
+					hasCodeChange = true
+					break
+				}
+			}
+
+			if !hasTestOrRetestCommand && !hasCodeChange {
```

Comment:
I'm not a big fan of defaulting to skip processing, it could fail unexpectedly in the future.

An alternative idea is something like:
```
var shouldSkipProcessing bool
for _, message := range currentMessages(change, lastUpdate) {
  if pjutil.RetestRe.Match([]byte(message.Message)) || pjutil.TestRe.Match([]byte(message.Message)) {
	  shouldSkipProcessing = false
	  break
  }
}

// check code change
...

if !shouldSkipProcessing {
  return nil;
}
```
the idea is that the for loop needs to be exercised so that the test can be skipped

Reply:
> since the zero value of `bool` is `false`, we have to set the initial value of `shouldSkipProcessing` to true anyway.

sorry but I disagree. The desired default should be a safe fallback, so even if everything is wrong:
```
var shouldSkipProcessing bool

// Complex logics with shouldSkipProcessing
...

if shouldSkipProcessing {
  // This code block should not be triggered even if the code above in the black box failed miserably
  return nil
}
```

As discussed offline that this PR is best effort to save time spent on inrepoconfig processing, so the bottom line is that if there is any bug in this logic then inrepoconfig should still be processed, and users won't tell the difference. The last thing we want to see is user updated a PR and Prow jobs not triggered because of bug here.

Reply:
@timwangmusic , please take a look at this comment

Reply:
Had one more offline discussion with Tim, realized that it's also ugly to implement the way that I proposed. It might be easier to move this short closure to a function for easier testing